### PR TITLE
[IMP] http: Sessions implicit deactivation

### PR DESCRIPTION
**Pull request by tbe-odoo**
- Store a token inside sessions to allow implicit session deactivation when needed.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: openerp/http.py [hunk 1]
```diff
@@ -1238,11 +1246,10 @@ def assert_valid(self, force=False):
 
         Ensures this session is valid (logged into the openerp server)
         """
-        if self.uid and not force:
+        if self.uid and self.session_token and not force:
```
Comment (tbe-odoo) [inline openerp/http.py]: Hello guys, I need your expertize on the modification of this method. I am not sure what are the consequences of this change.

In my knowledge their should never be a session with a login & db without the matching uid, meaning it was doing nothing ??

**Comment by tbe-odoo** [comment]
Should be merged with https://github.com/odoo/enterprise/pull/2245

**Comment by tbe-odoo** [comment]
Hello @dbo-odoo @mao-odoo , FYI 9.0 patch has been merged :)

**Comment by dbo-odoo** [comment]
@tbe-odoo @KangOl 
I did not think about our 260+ dbs still using `9.saas~11` on our saas, do you think it can be forward-ported to that version as-is?

**Pull request merged**